pkg/admission/prep_downscale.go

```diff
@@ -234,3 +251,12 @@ func deny(msg string, args ...any) *v1.AdmissionResponse {
 		},
 	}
 }
+
+func getResourceAnnotations(ctx context.Context, ar v1.AdmissionReview, api kubernetes.Interface) (map[string]string, error) {
+	switch ar.Request.Resource.Resource {
+	case "statefulsets":
+		obj, err := api.AppsV1().StatefulSets(ar.Request.Namespace).Get(ctx, ar.Request.Name, metav1.GetOptions{})
+		return obj.Annotations, err
```

Comment:
I believe `obj` will be nil if an error is returned:
```suggestion
		if err != nil {
				return nil, err
		}
		return obj.Annotations, nil
```